x2gbfs/providers/cambio.py

```diff
@@ -0,0 +1,149 @@
+    def load_vehicles(
+        self, default_last_reported: int
+    ) -> tuple[dict[str, dict[str, Any]] | None, dict[str, dict[str, Any]] | None]:
+        """
+        Returns vehicle_types_map (with a single vehicle type)
+        and vehicles_map, both keyed by the vehicle's/vehicle type's ID.
+        It's values must be dicts conforming to GBFS Spec v2.3 vehicles / vehicle types.
+        For details, see https://github.com/MobilityData/gbfs/blob/v2.3/gbfs.md
+        """
+
+        result = self._all_vehicle_types()
+
+        gbfs_vehicle_types_map: dict[str, dict[str, Any]] = {}
+        gbfs_vehicles_map: dict[str, dict[str, Any]] = {}
+        for elem in result:
+            # Note: usually you would iterate over a data source retrieve from a proprietary
+            # system and convert it to vehicles, reflectig theit real-time properties.
+            vehicle_type_id = elem['id']
+            propulsion_type = self._extract_propulsion_type(elem)
+
+            gbfs_vehicle_type = {
+                'vehicle_type_id': vehicle_type_id,
+                'name': elem['displayName'],
+                'form_factor': 'car',
+                'default_pricing_plan': elem['priceClass']['id'],
+                'propulsion_type': propulsion_type,
+                'max_range_meters': (
+                    self.DEFAULT_MAX_RANGE_ELECTRIC
+                    if propulsion_type == 'electric'
+                    else self.DEFAULT_MAX_RANGE_COMBUSTION
+                ),
+                'wheel_count': 4,
+                'return_constraint': 'roundtrip_station',
+            }
+            gbfs_vehicle_types_map[vehicle_type_id] = gbfs_vehicle_type
+
+        return gbfs_vehicle_types_map, gbfs_vehicles_map
+
+    def _extract_vehicle_types_available(self, elem: dict[str, Any]) -> list[dict[str, str | int]]:
+        """
+        Extracts a list of vehicle_types_available from station. As it is static,
+        we declare an availability of 1 per vehicle_type, though this incorrect.
+        """
+        vehicle_classes_at_station = elem.get('vehicleClasses', [])
+        return [{'vehicle_type_id': vehicle_class['id'], 'count': 1} for vehicle_class in vehicle_classes_at_station]
+
+    def _extract_propulsion_type(self, elem: dict[str, str]) -> str:
+        """
+        Guesses the propulsion type from vehicle name.
+        """
+        lowercase_name = elem['displayName'].lower()
+
+        if 'e-auto' in lowercase_name or 'smart ed' in lowercase_name:
+            return 'electric'
+        if 'transporter' in lowercase_name:
+            return 'combustion_diesel'
+
+        return 'combustion'
+
+    def _extract_station_rental_uris(self, elem: dict[str, str]) -> dict[str, str]:
+        """
+        Guesses the propulsion type from vehicle name.
+        """
+        station_name = unidecode_with_german_umlauts(elem['name'].lower())
+        station_url = f"https://www.cambio-carsharing.de/stationen/station/{station_name}-{elem['id']}"
```

Comment:
`station_name` and `elem['id']` are not getting URL-encoded. This would fail whenever any of the two contains a non-URL-safe character, e.g. `#`.